pkg/gds/gds_test.go

```diff
@@ -177,8 +177,135 @@ func (s *gdsTestSuite) TestRegister() {
 			Timestamp: sent,
 		},
 		{
-			Contact:   v.Contacts.Billing,
-			To:        v.Contacts.Billing.Email,
+			Contact:   v.Contacts.Legal,
+			To:        v.Contacts.Legal.Email,
+			From:      s.svc.GetConf().Email.ServiceEmail,
+			Subject:   emails.VerifyContactRE,
+			Reason:    "verify_contact",
+			Timestamp: sent,
+		},
+		{
+			Contact:   v.Contacts.Technical,
+			To:        v.Contacts.Technical.Email,
+			From:      s.svc.GetConf().Email.ServiceEmail,
+			Subject:   emails.VerifyContactRE,
+			Reason:    "verify_contact",
+			Timestamp: sent,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
+}
+
+func (s *gdsTestSuite) TestRegisterAlreadyVerified() {
```

Comment:
I think we were missing some coverage for the case where the VASP already has a verified contact so I added it.